code/src/weapons/Bullet/Bullet.tscn

```diff
@@ -0,0 +1,43 @@
+[gd_scene load_steps=3 format=2]
+
+[ext_resource path="res://src/weapons/Bullet/Bullet.gd" type="Script" id=1]
+
+[sub_resource type="CapsuleShape2D" id=1]
+
+custom_solver_bias = 0.0
+radius = 2.0
+height = 2.0
+
+[node name="Bullet" type="KinematicBody2D" index="0"]
```

Comment:
Why is this a `KinematicBody2D`?

Reply:
Because why not? :>

Do you think it'd better be an `Area2D`?

Reply:
>`KinematicBody2D` is for user-controlled things.

Not quite. `KinematicBody2D` is developer-, and not physics engine-controlled things. That is, the physics engine won't move a `KinematicBody2D` *at all*. It must be moved through scripts.

As it stands, bullets work. I see no benefit to switch to anything else.

We can change it later if the need arises.

Reply:
`KinematicBody2D` is a poor choice from a code-clarity perspective. It raises the question "why? How is the user controlling this?" -- it's meant to be used for player things that the player directly controls. `Area2D` makes more sense here because our bullets don't push stuff.

We should change this if/when we get a chance. This is not a priority though.

Reply:
I argue `KinematicBody2D` is as *good* choice as any, because as I've said, `KinematicBody2D` is for implementing bodies which are controlled from code -- code the devs write. The physics engine doesn't touch it. And I don't think it makes sense to have a body which doesn't interact with the physics engine which is meant to be used for things the player controls... Why can't `RigidBody2D` be used for the same purpose? Am I missing something?

Although I do think you have a point in terms of bullets not pushing stuff, `Area2D` isn't meant to be moved per se, either. Only way to move it is by directly altering its position, and that's flakey at best.